spec/games_spec.rb

```diff
@@ -0,0 +1,278 @@
+require 'rspec'
+require './lib/games'
+require './lib/game'
+
+game_uno = Game.new({
+  game_id:2012030221,
+  season:20122013,
+  type:"P",
+  date_time:2013-05-16,
+  away_team_id: "3",
+  home_team_id:"6",
+  away_goals:2,
+  home_goals:3,
+  outcome:"home win OT",
+  home_rink_side_starts:"left",
+  venue:"TD Garden",
+  venue_link:"/api/v1/venues/null",
+  venue_time_zone_id:"America/New_York",
+  venue_time_zone_offset:-4,
+  venue_time_zone_tz:"EDT"
+})
+
+game_dos = Game.new({
+  game_id:2016030132,
+  season:20162017,
+  type:"P",
+  date_time:2017-04-15,
+  away_team_id:"10",
+  home_team_id:"15",
+  away_goals:4,
+  home_goals:3,
+  outcome:"away win OT",
+  home_rink_side_starts:"right",
+  venue:"Verizon Center",
+  venue_link:"/api/v1/venues/null",
+  venue_time_zone_id:"America/New_York",
+  venue_time_zone_offset:-4,
+  venue_time_zone_tz:"EDT"
+})
+
+game_tres = Game.new({
+  game_id:2017030121,
+  season:20172018,
+  type:"P",
+  date_time:2018-04-12,
+  away_team_id:"10",
+  home_team_id:"6",
+  away_goals:1,
+  home_goals:5,
+  outcome:"home win REG",
+  home_rink_side_starts:"left",
+  venue:"Verizon Center",
+  venue_link:"/api/v1/venues/null",
+  venue_time_zone_id:"America/New_York",
+  venue_time_zone_offset:-4,
+  venue_time_zone_tz:"EDT"
+})
+
+game_quatro = Game.new({
+  game_id:2013020902,
+  season:20132014,
+  type:"R",
+  date_time:2014-03-01,
+  away_team_id:"15",
+  home_team_id:"6",
+  away_goals:4,
+  home_goals:2,
+  outcome:"away win REG",
+  home_rink_side_starts:"left",
+  venue:"TD Garden",
+  venue_link:"/api/v1/venues/null",
+  venue_time_zone_id:"America/New_York",
+  venue_time_zone_offset:-4,
+  venue_time_zone_tz:"EDT"
+})
+game_cinco = Game.new({
+  game_id:2012020592,
+  season:20122013,
+  type:"R",
+  date_time:2013-04-10,
+  away_team_id:"10",
+  home_team_id:"3",
+  away_goals:2,
+  home_goals:3,
+  outcome:"home win SO",
+  home_rink_side_starts:"right",
+  venue:"Madison Square Garden",
+  venue_link:"/api/v1/venues/null",
+  venue_time_zone_id:"America/New_York",
+  venue_time_zone_offset:-4,
+  venue_time_zone_tz:"EDT"
+})
+
+
+
+RSpec.describe Games do
```

Comment:
This is a massive improvement. Great job.

I think we can do even better by extracting a function that will be responsible for building games:

```
def default_game_params
  { game_id: 12345, season: 20122013, etc.}
end
```

```
def build_game(override_params = {})
  Game.new(default_game_params.merge(override_params))
end
```

And then override the values that we want to test in each test case:

```
it 'returns the highest total score' do
  games = [
    build_game({ game_id: 1, away_goals: 1, home_goals: 2 }),
    build_game({ game_id: 2, away_goals: 1, home_goals: 1 }),
  ]

  result = Games.highest_scoring_game(games)

  expect(result.game_id).to eq(2)
end
```